node/repo/fsrepo.go

```diff
@@ -284,10 +286,48 @@ func (fsr *fsLockedRepo) Datastore(ns string) (datastore.Batching, error) {
 }
 
 func (fsr *fsLockedRepo) Config() (interface{}, error) {
+	fsr.configLk.Lock()
+	defer fsr.configLk.Unlock()
+
+	return fsr.loadConfigFromDisk()
+}
+
+func (fsr *fsLockedRepo) loadConfigFromDisk() (interface{}, error) {
+	return config.FromFile(fsr.join(fsConfig), defConfForType(fsr.repoType))
+}
+
+func (fsr *fsLockedRepo) SetConfig(c func(interface{})) error {
 	if err := fsr.stillValid(); err != nil {
-		return nil, err
+		return err
 	}
-	return config.FromFile(fsr.join(fsConfig), defConfForType(fsr.repoType))
+
+	fsr.configLk.Lock()
+	defer fsr.configLk.Unlock()
+
+	cfg, err := fsr.loadConfigFromDisk()
+	if err != nil {
+		return err
+	}
+
+	// mutate in-memory representation of config
+	c(cfg)
+
+	// buffer into which we write TOML bytes
+	buf := new(bytes.Buffer)
+
+	// encode now-mutated config as TOML and write to buffer
+	err = toml.NewEncoder(buf).Encode(cfg)
```

Comment:
I'm not sure how to best do it - but by default we comment out all default values in the config - https://github.com/filecoin-project/lotus/blob/master/node/config/load.go#L45 - and ideally we'd keep default values commented out, but I can see this being rather tricky

Maybe @Kubuxu has some good idea on how to do this in a simple way, but if not, I'm fine with dropping comments on SetConfig, and fixing that later